ports/restbed/fix-cmake.patch

```diff
@@ -24,40 +21,29 @@ diff --git a/CMakeLists.txt b/CMakeLists.txt
  set_property( TARGET ${STATIC_LIBRARY_NAME} PROPERTY CXX_STANDARD 14 )
  set_property( TARGET ${STATIC_LIBRARY_NAME} PROPERTY CXX_STANDARD_REQUIRED ON )
  set_target_properties( ${STATIC_LIBRARY_NAME} PROPERTIES OUTPUT_NAME ${PROJECT_NAME} )
--	
++set ( LIBRARY_NAME "${STATIC_LIBRARY_NAME}")
 +else ( )
+ 	
  set( SHARED_LIBRARY_NAME "${PROJECT_NAME}-shared" )
  add_library( ${SHARED_LIBRARY_NAME} SHARED ${MANIFEST} )
- set_property( TARGET ${SHARED_LIBRARY_NAME} PROPERTY CXX_STANDARD 14 )
-@@ -93,13 +94,20 @@ else ( )
+@@ -93,8 +96,13 @@ else ( )
  	set_target_properties( ${SHARED_LIBRARY_NAME} PROPERTIES OUTPUT_NAME ${PROJECT_NAME} )
  endif ( )	
  set_target_properties( ${SHARED_LIBRARY_NAME} PROPERTIES SOVERSION ${PROJECT_VERSION_MAJOR} VERSION ${PROJECT_VERSION_MAJOR}.${PROJECT_VERSION_MINOR} )
++set ( LIBRARY_NAME "${SHARED_LIBRARY_NAME}")
 +endif ( )
  
  if ( BUILD_SSL )
-+if ( BUILD_SHARED_LIBS )
++    target_link_libraries( ${LIBRARY_NAME} LINK_PRIVATE OpenSSL::SSL)
```

Comment:
The original code also linked to crypto_LIBRARY_SHARED/crypto_LIBRARY_STATIC which seem to map to OpenSSL::Crypto. Are you sure we don't need to link to it? Also, what about CMAKE_DL_LIBS that the original code links to in the static library case?

Reply:
I compiled at least on one static triplet before submitting.
I assume crypto_LIBRARY_* and CMAKE_DL_LIBS are just transitive usage requirements of the main openssl lib.
Target OpenSSL::SSL depends on OpenSSL::Crypto and on whatever CMAKE_DL_LIBS actually is. That's the key benefit of imported targets if they are done correctly: Consumers (here: restbed) don't need to know and repeat the transitive usage requirements.
Things look different if restbed actually used libcrypto or dl directly.